python/samples/concepts/auto_function_calling/nexus_raven.py

```diff
@@ -0,0 +1,359 @@
+    OpenAIChatCompletion,
+)
+from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+    OpenAIChatPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.connectors.ai.text_completion_client_base import TextCompletionClientBase
+from semantic_kernel.contents import (
+    ChatHistory,
+    ChatMessageContent,
+    FunctionCallContent,
+    StreamingChatMessageContent,
+    TextContent,
+)
+from semantic_kernel.contents.function_result_content import FunctionResultContent
+from semantic_kernel.functions import KernelArguments, kernel_function
+
+kernel = Kernel()
+
+#########################################################################
+# Step 0: Define a custom AI Service, with Prompt Execution settings. ###
+# This uses huggingface_hub package, so install that if needed.       ###
+#########################################################################
+
+
+class NexusRavenPromptExecutionSettings(PromptExecutionSettings):
+    do_sample: bool = True
+    max_new_tokens: int | None = None
+    stop_sequences: Any = None
+    temperature: float | None = None
+    top_p: float | None = None
+
+    def prepare_settings_dict(self, **kwargs) -> dict[str, Any]:
+        """Prepare the settings dictionary."""
+        return self.model_dump(
+            include={"max_new_tokens", "temperature", "top_p", "do_sample", "stop_sequences"},
+            exclude_unset=False,
+            exclude_none=True,
+            by_alias=True,
+        )
+
+
+class NexusRavenCompletion(TextCompletionClientBase, ChatCompletionClientBase):
+    """To use this class, you should have installed the ``huggingface_hub`` package, and
+    the environment variable ``HUGGINGFACEHUB_API_TOKEN`` set with your API token,
+    or given as a named parameter to the constructor."""
+
+    client: AsyncInferenceClient
+
+    def __init__(
+        self,
+        service_id: str,
+        ai_model_id: str,
+        endpoint_url: HttpUrl,
+        api_token: str | None = None,
+        client: AsyncInferenceClient | None = None,
+    ):
+        if not client:
+            client = AsyncInferenceClient(model=endpoint_url, token=api_token)
+        super().__init__(service_id=service_id, ai_model_id=ai_model_id, client=client)
+
+    async def get_chat_message_contents(
+        self,
+        chat_history: "ChatHistory",
+        settings: "PromptExecutionSettings",
+        **kwargs: Any,
+    ) -> list["ChatMessageContent"]:
+        """Creates a chat message content with a function call content within.
+
+        Uses the text content and and parses it into a function call content."""
+        result = await self.get_text_contents(
+            prompt=chat_history.messages[-1].content,
+            settings=settings,
+        )
+        fcc, frc = await self._execute_function_calls(result[0], **kwargs)
```

Comment:
As a sample, what do you think about making these names for explicit? Some users may not know what `fcc` or `frc` stand for.